File: array/283_move_zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slow = fast = 0

        while fast < len(nums):
            if nums[fast] != 0:
                nums[slow], nums[fast] = nums[fast], nums[slow]
                slow += 1
            fast += 1
        # index = 0

        # for i in range(len(nums)):
        #     if nums[i] != 0:
        #         nums[index] = nums[i]
        #         index += 1
            
        # for j in range(index, len(nums)):
        #     nums[j] = 0
        

File: two_pointer/283_move_zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int k = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                continue;
            }
            nums[k++] = nums[i];
        }
        for (; k < nums.length; k++) {
            nums[k] = 0;
        }
    }
}